vignettes/benchmarks.Rmd.orig

```diff
@@ -0,0 +1,914 @@
+    stan = stan_opts(method = "laplace", backend = "cmdstanr")
+  ),
+  # The 7-day RW Rt model with MCMC fitting
+  rw7_mcmc = list(
+    rt = rt_opts(
+      prior = rt_prior_default,
+      rw = 7
+    )
+  ),
+  # The 7-day RW Rt model with variational bayes fitting
+  rw7_vb = list(
+    rt = rt_opts(
+      prior = rt_prior_default,
+      rw = 7
+    ),
+    stan = stan_opts(method = "vb", backend = "rstan")
+  ),
+  # The 7-day RW Rt model with pathfinder fitting
+  rw7_pathfinder = list(
+    rt = rt_opts(
+      prior = rt_prior_default,
+      rw = 7
+    ),
+    stan = stan_opts(method = "pathfinder", backend = "cmdstanr")
+  ),
+  # The 7-day RW Rt model with laplace fitting
+  rw7_laplace = list(
+    rt = rt_opts(
+      prior = rt_prior_default,
+      rw = 7
+    ),
+    stan = stan_opts(method = "laplace", backend = "cmdstanr")
+  ),
+  # The non_residual model with MCMC fitting
+  non_residual_mcmc = list(
+    rt = rt_opts(
+      prior = rt_prior_default,
+      gp_on = "R0"
+    )
+  ),
+  # The non_residual model with variational bayes fitting
+  non_residual_vb = list(
+    rt = rt_opts(
+      prior = rt_prior_default,
+      gp_on = "R0"
+    ),
+    stan = stan_opts(method = "vb", backend = "rstan")
+  ),
+  # The non_residual model with pathfinder fitting
+  non_residual_pathfinder = list(
+    rt = rt_opts(
+      prior = rt_prior_default,
+      gp_on = "R0"
+    ),
+    stan = stan_opts(method = "pathfinder", backend = "cmdstanr")
+  ),
+  # The non_residual model with laplace fitting
+  non_residual_laplace = list(
+    rt = rt_opts(
+      prior = rt_prior_default,
+      gp_on = "R0"
+    ),
+    stan = stan_opts(method = "pathfinder", backend = "cmdstanr")
+  )
+)
+```
+
+## Running the models
+
+All the models will share the configuration for the generation time, incubation period, reporting delay, and the forecast horizon, so we will define them once and pass them to the models.
+
+```{r, constant_inputs}
+# Generation time
+generation_time <- Gamma(
+  shape = Normal(1.3, 0.3),
+  rate = Normal(0.37, 0.09),
+  max = 14
+)
+
+# Incubation period
+incubation_period <- LogNormal(
+  meanlog = Normal(1.6, 0.05),
+  sdlog = Normal(0.5, 0.05),
+  max = 14
+)
+
+# Reporting delay
+reporting_delay <- LogNormal(
+  meanlog = 0.5,
+  sdlog = 0.5,
+  max = 10
+)
+
+# Combine the incubation period and reporting delay into one delay
+delay <- incubation_period + reporting_delay
+
+# 7-day forecast window
+horizon <- 7
+
+# Combine the shared model inputs into a list for use across all the models
+model_inputs <- list(
+  generation_time = generation_time_opts(generation_time),
+  delays = delay_opts(delay),
+  obs = obs,
+  forecast = forecast_opts(horizon = horizon),
+  verbose = FALSE
+)
+```
+
+Additionally, from the benchmarking data, we choose the following dates to represent the periods of growth, peak, and decline.
+```{r snapshot_dates}
+snapshot_dates <- as.Date(c("2020-05-15", "2020-06-21", "2020-06-28"))
```

Comment:
Might be good to show a plot of R with these highlighted with vertical lines - looking at that plot I think I would have located these dates elsewhere (e.g. the "decline" one is only just starting to decline).

Reply:
Addressed in 09d46e076aed7dccd05d5a596a39cc1a4537b8f2.